mvc/controllers/actions.rst

```diff
@@ -1,8 +1,63 @@
-.. include:: /../common/stub-topic.txt
+Actions and Action Results
+==========================
 
-|stub-icon| Actions and Action Results
-===========================================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+Actions and action results are a fundamental part of how developers build apps using ASP.NET MVC.
 
-.. _issue: https://github.com/aspnet/Docs/issues/118
+Sections:
+	- `What is a Controller`_
+	- `Defining Actions`_
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/1.0.0-rc1/mvc/performance/response-caching/sample>`_.
+
+What is a Controller
+--------------------
+
+In ASP.NET MVC, a `Controller` is used to define and group a set of actions. An `action` (or `action method`) is a method on a controller that handles incoming requests. In ASP.NET 5, a controller can be any instantiable class that ends in "Controller" or inherits from a class that ends with "Controller". By convention, controller classes may be placed within a root-level "Controllers" folder within the app, and may inherit from ``Microsoft.AspNet.Mvc.Controller``, but these are not required.
+
+Controllers support :doc:`dependency injection <dependency-injection>`, and should follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle>`_ and request any dependencies their actions require through their constructor. Controllers provide a logical means of grouping similar actions together, allowing common sets of rules (e.g. routing, caching, authorization) to be applied collectively.
+
+Within the Model-View-Controller pattern, a Controller is responsible for the initial processing of the request and instantiation of the Model. Business decisions should ideally be performed within the Model. The Controller then takes the result of the Model's processing (if any), chooses which View should be returned, and returns that View along with any data that View requires. Learn more: :doc:`../overview`.
+
+.. tip:: The Controller is a `UI level` abstraction. Its responsibility is to ensure incoming request data is valid and to choose which View (or result) should be returned. In well-factored apps it will not include data access or business logic, but instead will delegate to services handling these responsibilities.
+ 
+Defining Actions
+----------------
+Any public method on a controller type is an action. Parameters on actions are bound to request data and validated using :doc:`model binding <../models/model-binding>`. Any action that accepts parameters should verify that the incoming data is valid by checking ``ModelState.IsValid``.
```

Comment:
Updated with all feedback. Main purpose of Tip is to remind developers not to put data access and business logic in their actions/controllers. I'm happy to link somewhere for more info but am not sure where or what more it would say.